### PR TITLE
Bug 1822620 -A possible fix for NullPointerException.

### PR DESCRIPTION
**Pull request by iorgamgabriel**
This may be a possible fix for this crash: https://crash-stats.mozilla.org/report/index/a42f6990-a965-4578-b56d-ed4870230315 
I can't reproduce the bug. 

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1822620

### CODE REVIEW

**Review by gabrielluong — APPROVED** [review]
@iorgamgabriel Let's make the commit message more descriptive here - `Bug 1822620 - Check if the fragment is still attached when calling updateDeleteButton in DeleteBrowsingDataFragment`

What I suspect is happening here is that the user is navigating out of the DeleteBrowsingDataFragment after pressing the delete browsing data button. In https://github.com/mozilla-mobile/fenix/pull/28672, we disable the buttons while the deletion is in progress, but the user could navigate away and finishDeletion is called.

File: fenix/app/src/main/java/org/mozilla/fenix/settings/deletebrowsingdata/DeleteBrowsingDataFragment.kt [hunk 1]
```diff
@@ -115,10 +115,12 @@ class DeleteBrowsingDataFragment : Fragment(R.layout.fragment_delete_browsing_da
     }
 
     private fun updateDeleteButton(deleteInProgress: Boolean = false) {
-        val enabled = getCheckboxes().any { it.isChecked } && !deleteInProgress
+        context?.let {
```
Comment (gabrielluong) [inline fenix/app/src/main/java/org/mozilla/fenix/settings/deletebrowsingdata/DeleteBrowsingDataFragment.kt:118]: Perhaps consider changing all of this to use `runIfFragmentIsAttached` https://searchfox.org/mozilla-mobile/source/firefox-android/fenix/app/src/main/java/org/mozilla/fenix/ext/Fragment.kt#52

Comment (iorgamgabriel) [reply fenix/app/src/main/java/org/mozilla/fenix/settings/deletebrowsingdata/DeleteBrowsingDataFragment.kt]: Done

**Comment by iorgamgabriel** [comment]
@mergify backport releases_v112

**Pull request merged**